### DISCUSSION TITLE
Refactoring not working with Gauge IntelliJ Plugin

### DISCUSSION THREAD

**Discussion by preetipadalia** [getgauge/gauge#1886] [General]
Hi,

Getting following error with gauge Intellij plugin while refactoring the step:

```
Exception occurred in Gauge plugin
				More details...
				Please log an issue in https://github.com/getgauge/intellij-plugin with following details:
				#### gauge process exited with code 2
				```
				panic: runtime error: invalid memory address or nil pointer dereference
				[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x1609596]
				
				goroutine 37 [running]:
				github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).performRefactoring.func1(0x0, 0x0)
					/home/runner/work/gauge/gauge/api/apiMessageHandler.go:189 +0x26
				github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).performRefactoring(0xc0002ca040, 0xc000010000, 0xc000240000)
					/home/runner/work/gauge/gauge/api/apiMessageHandler.go:197 +0x4ce
				github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).MessageBytesReceived(0xc0002ca040, 0xc0003d2209, 0x84, 0xb9, 0x1989800, 0xc000276040)
					/home/runner/work/gauge/gauge/api/apiMessageHandler.go:61 +0x45b
				github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).processMessage(0xc0002ca080, 0xc000115f90, 0x1989800, ... (show balloon)

```
Thanks
Preeti

**Comment by preetipadalia** [comment]
Hi,
Any input on this?

Thanks
Preeti

**Reply by zabil** [reply]
Unfortunately there are not enough details in your report to debug the issue. Please specify version information and please upgrade everything to the latest version and try.

**Reply by preetipadalia** [reply]
Hi zabil please find the details below:

IntelliJ Version:
-------
IntelliJ IDEA 2019.3.4 (Ultimate Edition)
Build #IU-193.6911.18, built on March 17, 2020
Runtime version: 11.0.6+8-b520.43 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains


Gauge Version:
-------
Gauge version: 1.1.7
Commit Hash: 5d86b72

Plugins
-------
html-report (4.0.12)
java (0.7.13)
js (2.3.15)
reportportal (1.11.1)
screenshot (0.0.1)

OS: 
-------
macOS Catalina 10.15.7

java -version
--------
openjdk version "11.0.10" 2021-01-19 LTS
OpenJDK Runtime Environment Corretto-11.0.10.9.1 (build 11.0.10+9-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.10.9.1 (build 11.0.10+9-LTS, mixed mode)

**Gauge project is built using Maven.

Thanks
Preeti

**Reply by zabil** [reply]
> IntelliJ IDEA 2019.3.4 (Ultimate Edition)
Build #IU-193.6911.18, built on March 17, 2020

This version of IntelliJ looks out of date. It's probably not downloading the latest version of the Gauge IntellIj plugin. Please upgrade this version and make sure that the Gauge IntelliJ plugin uses the latest [version](https://plugins.jetbrains.com/plugin/7535-gauge)

**Reply by preetipadalia** [reply]
It worked after IntelliJ Upgrade. 

Thank you so much @zabil  for your help. :)

**Reply by preetipadalia** [reply]
@zabil it worked for the first time and now again it started giving issues.

IntelliJ Version
------------
IntelliJ IDEA 2020.3.2 (Ultimate Edition)
Build #IU-203.7148.57, built on January 26, 2021
Runtime version: 11.0.9.1+11-b1145.77 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 10.15.7
Non-Bundled Plugins: com.thoughtworks.gauge

**Reply by preetipadalia** [reply]
More details...  Please report an issue in https://youtrack.jetbrains.com/issues/IDEA with following details:  
#### Gauge process exited with code ```2``` 
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x1609596]  
goroutine 66 [running]: 
github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).performRefactoring.func1(0x0, 0x0) 	/home/runner/work/gauge/gauge/api/apiMessageHandler.go:189 +0x26 github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).performRefactoring(0xc000372000, 0xc00001a2d0, 0xc00021e000) 	/home/runner/work/gauge/gauge/api/apiMessageHandler.go:197 +0x4ce
 github.com/getgauge/gauge/api.(*gaugeAPIMessageHandler).MessageBytesReceived(0xc000372000, 0xc0003b80d1, 0x36, 0xc4, 0x1989800, 0xc000394008) 	
/home/runner/work/gauge/gauge/api/apiMessageHandler.go:61 +0x45b 
github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).processMessage(0xc000372040, 0xc000199f90, 0x1989800, 0xc000394008) 	
/home/runner/work/gauge/gauge/conn/connectionHandler.go:122 +0x12e 
github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).handleConnectionMessages(0xc000372040, 0x1989800, 0xc000394008) 	
/home/runner/work/gauge/gauge/conn/connectionHandler.go:112 +0xb4 created by github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).acceptConnectionWithoutTimeout 	/home/runner/work/gauge/gauge/conn/connectionHandler.go:87 +0x1cb  
Gauge daemon initialized and listening on port: 49407 
exec: "mvn": executable file not found in $PATH I
nstalled version of gauge-java (0.7.13) does not match with dependency gauge-java () specified in pom.xml file 
No Java runtime present, requesting install. 
Error occurred while waiting for runner process to finish. Error : exit status 1 * IDE version: 2020.3.2 * API version: IU-203.7148.57 * Plugin version: 203.5981.155

**Reply by zabil** [reply]
To be honest. I am not able to replicate this consistently. Make sure you are also using the latest version of the Gauge maven plugin in the pom.xml file.

Sadly, this part Gauge + IntelliJ is not maintained anymore and it is unlikely there will be a fix anytime soon. Sorry for that.

**Reply by preetipadalia** [reply]
Oh, is there a way I can check or debug the issue?

**Reply by preetipadalia** [reply]
It seems IntelliJ plugin was not identifying the installed java. After I installed Oracle jdk it started working.
